operator/v1/types_dns.go

```diff
@@ -287,6 +287,17 @@ type ForwardPlugin struct {
 	//
 	// +optional
 	TransportConfig DNSTransportConfig `json:"transportConfig,omitempty"`
+
+	// protocolPolicy configures the protocol policy for upstream DNS requests.
```

Comment:
```suggestion
	// protocol configures the protocol used by requests to upstream DNS.
```

Reply:
I would also add something about what happens if you try TCP and it isn't really needed, and the reasons why you would try TCP (mostly for non-compliant DNS servers, iiuc).

Reply:
Good idea, I added a sentence on why it's useful and what the impacts may be.